src/omt/omt_optimizer.cpp

```diff
@@ -39,32 +42,143 @@ std::unique_ptr<OMTOptimizer> OMTOptimizer::getOptimizerForNode(TNode targetNode
   else if (objectiveType.isBitVector())
   {
     // bitvector type: use OMTOptimizerBitVector
-    return std::unique_ptr<OMTOptimizer>(new OMTOptimizerBitVector(isSigned));
+    return std::unique_ptr<OMTOptimizer>(
+        new OMTOptimizerBitVector(objective.bvIsSigned()));
   }
   else
   {
     return nullptr;
   }
 }
 
-std::unique_ptr<SmtEngine> OMTOptimizer::createOptCheckerWithTimeout(
-    SmtEngine* parentSMTSolver, bool needsTimeout, unsigned long timeout)
+Node OMTOptimizer::mkStrongIncrementalExpression(
+    NodeManager* nm, TNode lhs, TNode rhs, OptimizationObjective& objective)
+{
+  constexpr const char lhsTypeError[] =
+      "lhs type does not match or is not implicitly convertable to the target "
+      "type";
+  constexpr const char rhsTypeError[] =
+      "rhs type does not match or is not implicitly convertable to the target "
+      "type";
+  TypeNode targetType = objective.getTarget().getType();
+  switch (objective.getType())
+  {
+    case OptimizationObjective::MINIMIZE:
+    {
+      if (targetType.isInteger())
+      {
+        Assert(lhs.getType().isInteger() || lhs.getType().isReal())
+            << lhsTypeError;
+        Assert(rhs.getType().isInteger() || rhs.getType().isReal())
```

Comment:
Please remove the `isReal()` from these places as well (similarly for the similar places  below e.g. for maximize).

Reply:
this is desired, because Real numbers are eligible to use LT and GT and they could be compared against integers. 
Not restricted to Integer.

Reply:
but in the current code, is there any benchmark that will make `lhs.getType()` to be a real? Aren't we guarding to get only bit-vectors and integers?

Reply:
The function is supposed to make an incremental expression if the type is same as obj.getTarget() or is implicitly convertible to that... So it's desired behavior. 
test cases doesn't reflect this.

Reply:
If the module currently only supports integers and bit-vectors, then handling reals is confusing and redundant. If it is not redundant then please add a unit test that makes `rhs.getType().isReal()` so that this case is properly tested. Otherwise, please remove.